### DISCUSSION TITLE
ERROR RESTY http: server gave HTTP response to HTTPS client

### DISCUSSION THREAD

**Discussion by glynhudson** [shellhub-io/shellhub#4199] [Q&A]
Hi, I'm running a self-hosted shellhub V0.16.0. All my existing devices work fine and I can add some new devices, but when I try to add other devices I get the error `ERROR RESTY http: server gave HTTP response to HTTPS client` in docker logs on the client device. 

The shellhub server is fully https enabled with http disabled, therefore I'm unsure when the HTTP response is coming from.

The client with the issue is running docker `Docker version 27.3.1,`

Any ideas? Thanks.

**Comment by gustavosbarreto** [comment]
What is the value of the `SERVER_ADDRESS` environment variable in the agent's container? Does the address start with `http` or `https`?

**Comment by glynhudson** [comment]
How can I check the environment variables inside the shellhub agent container? 

It doesn't seem to be possible to attach a shell to the agent container? I.e running  `docker exec -it shellhub /bin/bash`  returns `OCI runtime exec failed: exec failed: unable to start container process: exec: "/bin/bash": stat /bin/bash: no such file or directory: unknown`

**Reply by otavio** [reply]
You can use `docker inspect` command. This will provide a general overview of what configuration the image is using.
